dm/worker/server.go

```diff
@@ -706,6 +708,12 @@ func (s *Server) startWorker(cfg *config.MysqlConfig) error {
 		return err
 	}
 	s.setWorker(w, false)
+	defer func() {
+		// close w when some error occurs
+		if err1 != nil {
+			w.Close()
```

Comment:
`Close` worker when starting any subtask failed?

Reply:
If we start subtask failed, shouldn't we free this worker for other sources?